tests/protractor/priority/priority.spec.js

```diff
@@ -18,49 +18,45 @@ describe('Priority App', function () {
   }
 
   function sleep() {
-    return flow.execute(waitOne);
+    flow.execute(waitOne);
   }
 
-  beforeEach(function () {
-    if( !firebaseCleared ) {
-      firebaseCleared = true;
-      flow.execute(purge);
-    }
-
-    // Navigate to the priority app
-    browser.get('priority/priority.html');
+  function clearFirebaseRef() {
+    var deferred = protractor.promise.defer();
 
-    // Wait for all data to load into the client
-    flow.execute(waitForData);
+    firebaseRef.remove(function(err) {
+      if (err) {
+        deferred.reject(err);
+      } else {
+        deferred.fulfill();
+      }
+    });
 
-    function purge() {
-      var def = protractor.promise.defer();
-      firebaseRef.remove(function(err) {
-        if( err ) { def.reject(err); }
-        else { def.fulfill(true); }
-      });
-      return def.promise;
-    }
+    return deferred.promise;
+  }
 
-    function waitForData() {
-      var def = protractor.promise.defer();
-      firebaseRef.once('value', function() {
-        waitOne().then(function() {
-          def.fulfill(true);
-        });
-      });
-      return def.promise;
+  beforeEach(function (done) {
+    if (!isPageLoaded) {
+      // Navigate to the priority app
+      browser.get('priority/priority.html').then(function() {
+        isPageLoaded = true;
```

Comment:
Is this the right place for isPageLoaded? It looks like browser.get() could run multiple times before this flag is set.